### PR TITLE
Enable ASTC encoder build when building with ANGLE.

### LINKED ISSUE #94814: Crash on macOS when importing textures with specific settings
### Tested versions

4.3-rc1 (also happens in 4.3-beta3)

### System information

macOS Sonoma (14.5) with Intel CPU - GPU: Radeon Pro Vega 20

### Issue description

When opening the editor to import textures, the editor crashes with a segfault. I'm using the Compatibility renderer.

<details>
<summary>Stack trace</summary>

```
* thread #1, queue = 'com.apple.main-thread', stop reason = signal SIGABRT
  * frame #0: 0x00007ff810a4b14a libsystem_kernel.dylib`__pthread_kill + 10
    frame #1: 0x0000000107c57555 libsystem_pthread.dylib`pthread_kill + 262
    frame #2: 0x00007ff8109a9b19 libsystem_c.dylib`__abort + 145
    frame #3: 0x00007ff810997ab8 libsystem_c.dylib`__stack_chk_fail + 102
    frame #4: 0x00000001009d60d8 Godot`_compress_astc(Image*, Image::ASTCFormat) + 2168
    frame #5: 0x00000001046031c8 Godot`Image::compress_from_channels(Image::CompressMode, Image::UsedChannels, Image::ASTCFormat) + 200
    frame #6: 0x0000000102093ca5 Godot`ResourceImporterTexture::save_to_ctex_format(Ref<FileAccess>, Ref<Image> const&, ResourceImporterTexture::CompressMode, Image::UsedChannels, Image::CompressMode, float) + 837
    frame #7: 0x00000001020944b1 Godot`ResourceImporterTexture::_save_ctex(Ref<Image> const&, String const&, ResourceImporterTexture::CompressMode, float, Image::CompressMode, bool, bool, bool, bool, bool, bool, bool, bool, unsigned int, Ref<Image> const&, Image::RoughnessChannel) + 721
    frame #8: 0x00000001020960e1 Godot`ResourceImporterTexture::import(String const&, String const&, HashMap<StringName, Variant, HashMapHasherDefault, HashMapComparatorDefault<StringName>, DefaultTypedAllocator<HashMapElement<StringName, Variant>>> const&, List<String, DefaultAllocator>*, List<String, DefaultAllocator>*, Variant*) + 6257
    frame #9: 0x0000000101b78722 Godot`EditorFileSystem::_reimport_file(String const&, HashMap<StringName, Variant, HashMapHasherDefault, HashMapComparatorDefault<StringName>, DefaultTypedAllocator<HashMapElement<StringName, Variant>>> const&, String const&, Variant*) + 6178
    frame #10: 0x0000000101b6afe3 Godot`EditorFileSystem::reimport_files(Vector<String> const&) + 3091
    frame #11: 0x0000000101b68b8e Godot`EditorFileSystem::_update_scan_actions() + 5854
    frame #12: 0x0000000101b6f13d Godot`EditorFileSystem::_notification(int) + 621
    frame #13: 0x0000000104a469de Godot`Object::notification(int, bool) + 62
    frame #14: 0x0000000102898bb5 Godot`SceneTree::_process_group(SceneTree::ProcessGroup*, bool) + 405
    frame #15: 0x0000000102896ddf Godot`SceneTree::_process(bool) + 879
    frame #16: 0x000000010289748f Godot`SceneTree::process(double) + 223
    frame #17: 0x0000000100971f71 Godot`Main::iteration() + 1089
    frame #18: 0x000000010090314a Godot`OS_MacOS::run() + 154
    frame #19: 0x00000001009321f3 Godot`main + 387
    frame #20: 0x00007ff8106f8366 dyld`start + 1942
```

</details>

Only happens on macOS. On Linux and Windows it imports the texture just fine. 

Disabling ETC2 ASTC makes it not crash but it does not import the texture correctly.

### Steps to reproduce

- Create a `project.godot` with these settings:
```ini
[importer_defaults]

texture={
"compress/channel_pack": 0,
"compress/hdr_compression": 1,
"compress/high_quality": true,
"compress/lossy_quality": 0.7,
"compress/mode": 2,
"compress/normal_map": 0,
"detect_3d/compress_to": 1,
"editor/convert_colors_with_editor_theme": false,
"editor/scale_with_editor_scale": false,
"mipmaps/generate": false,
"mipmaps/limit": -1,
"process/fix_alpha_border": true,
"process/hdr_as_srgb": false,
"process/hdr_clamp_exposure": false,
"process/normal_map_invert_y": false,
"process/premult_alpha": false,
"process/size_limit": 0,
"roughness/mode": 0,
"roughness/src_normal": "",
"svg/scale": 1.0
}

[rendering]

renderer/rendering_method="gl_compatibility"
renderer/rendering_method.mobile="gl_compatibility"
textures/vram_compression/import_etc2_astc=true
```

- Include any texture in the project (can be the default `icon.svg`).
- Open the editor to import the files.

### Minimal reproduction project (MRP)

[import-crash-macos.zip](https://github.com/user-attachments/files/16395211/import-crash-macos.zip)

### PR DESCRIPTION
**Pull request by bruvzg**
Depends on https://github.com/godotengine/godot-angle-static/pull/13
Fixes https://github.com/godotengine/godot/issues/94814

### CODE REVIEW

**Review by fire — APPROVED** [review]

File: modules/astcenc/config.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 def can_build(env, platform):
-    return env.editor_build
+    return env.editor_build or env.get("angle_libs")
```
Comment (akien-mga) [inline modules/astcenc/config.py:2]: Might warrant a comment.

We don't use the astcenc in non-editor builds as it's only used in the editor to encode textures. I assume ANGLE needs it in export templates too to re-encode some stuff on the fly?

Comment (bruvzg) [reply modules/astcenc/config.py:2]: Yes, ANGLE always needs it (previously was part of ANGLE lib).

Comment (akien-mga) [inline modules/astcenc/config.py:2]:
```suggestion
    # Godot only uses it in the editor, but ANGLE depends on it and we had
    # to remove the copy from prebuilt ANGLE libs to solve symbol clashes.
    return env.editor_build or env.get("angle_libs")
```

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**